src/rootfstreemanager.cc

```diff
@@ -83,10 +83,18 @@ data::InstallationResult RootfsTreeManager::install(const Uptane::Target& target
   Uptane::Target current = OstreeManager::getCurrent();
   // Do ostree install if the currently installed target's hash differs from the specified target's hash,
   // or there is pending installation and it differs from the specified target so we undeploy it and make the new target
-  // pending
+  // pending (app driven rollback)
   if ((current.sha256Hash() != target.sha256Hash()) ||
       (!sysroot()->getDeploymentHash(OSTree::Sysroot::Deployment::kPending).empty() &&
        sysroot()->getDeploymentHash(OSTree::Sysroot::Deployment::kPending) != target.sha256Hash())) {
+    // If the boot fw update is in progress and it is an installation of Target ostree of which differs
+    //  from the ostree a device is booted on then
+    if (boot_fw_update_status_->isUpdateInProgress() && (current.sha256Hash() != target.sha256Hash())) {
+      LOG_WARNING << "Boot fw update is in progress."
+                     " A device must be rebooted to confirm and finalize the boot fw update"
+                     " before installation of a new Target with ostree/rootfs change";
+      return data::InstallationResult(data::ResultCode::Numeric::kNeedCompletion, "");
```

Comment:
doesn't our main loop interpret this response as "install went well, reboot now"?

Reply:
Yep, that's exactly what we need in this case.

Reply:
We need to reboot twice. At first to confirm and finalize ostree. And then before installation of another new ostree version we need to do another reboot to confirm and finalize boot fw update.

Reply:
is this going to trick our backend into thinking the device is about install this new target and not that it needs to do something for the previous one?

Reply:
the previous one is already installed.
aklite will install this new Target just after reboot.

Reply:
I just would like to get confirmation from @igoropaniuk that this idea works at all, before maturing it.

Reply:
i think your explanation makes sense. i think maybe just add some comment text to help explain.